src/layout.rs

```diff
@@ -0,0 +1,295 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANtIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use std::iter::Iterator;
+
+use hashbrown::{HashMap, HashSet};
+
+use pyo3::prelude::*;
+
+use petgraph::graph::NodeIndex;
+use petgraph::prelude::*;
+use petgraph::EdgeType;
+
+type Nt = f64;
```

Comment:
well, not really. my initial thought was that this could be an "adjustable" numerical precision. do you feel that i should explicitly use f64 in the code?

Reply:
It was more as I was reading the module I kind of skipped past the type definition at first and saw everything taking `Nt` where I would have expected `f64` and didn't know what the reason for aliasing it was. It's not a big deal I personally probably would have stuck with `f64` everywhere. But it's just code churn at this point to find replace `Nt` with `f64` so it's probably not worth updating.